src/core/mount-setup.c

```diff
@@ -229,76 +229,105 @@ int mount_setup_early(void) {
+        if (r < 0)
+                return log_error_errno(r, "Failed to create symlink %s: %m", a);
+
+#ifdef SMACK_RUN_LABEL
+        const char *p;
+
+        p = strjoina("/sys/fs/cgroup/", target);
+
+        r = mac_smack_copy(a, p);
+        if (r < 0 && r != -EOPNOTSUPP)
+                return log_error_errno(r, "Failed to copy smack label from %s to %s: %m", p, a);
+#endif
+
+        return 0;
+}
+
+int mount_cgroup_controllers(void) {
         _cleanup_set_free_free_ Set *controllers = NULL;
-        bool has_argument = !!join_controllers;
         int r;
 
         if (!cg_is_legacy_wanted())
                 return 0;
 
         /* Mount all available cgroup controllers that are built into the kernel. */
-
-        if (!has_argument)
-                /* The defaults:
-                 * mount "cpu" + "cpuacct" together, and "net_cls" + "net_prio".
-                 *
-                 * We'd like to add "cpuset" to the mix, but "cpuset" doesn't really
-                 * work for groups with no initialized attributes.
-                 */
-                join_controllers = (char**[]) {
-                        STRV_MAKE("cpu", "cpuacct"),
-                        STRV_MAKE("net_cls", "net_prio"),
-                        NULL,
-                };
-
         r = cg_kernel_controllers(&controllers);
         if (r < 0)
                 return log_error_errno(r, "Failed to enumerate cgroup controllers: %m");
 
         for (;;) {
                 _cleanup_free_ char *options = NULL, *controller = NULL, *where = NULL;
+                const char *other_controller;
                 MountPoint p = {
                         .what = "cgroup",
                         .type = "cgroup",
                         .flags = MS_NOSUID|MS_NOEXEC|MS_NODEV,
                         .mode = MNT_IN_CONTAINER,
                 };
-                char ***k = NULL;
 
                 controller = set_steal_first(controllers);
                 if (!controller)
                         break;
 
-                for (k = join_controllers; *k; k++)
-                        if (strv_find(*k, controller))
-                                break;
-
-                if (*k) {
-                        char **i, **j;
-
-                        for (i = *k, j = *k; *i; i++) {
-
-                                if (!streq(*i, controller)) {
-                                        _cleanup_free_ char *t;
-
-                                        t = set_remove(controllers, *i);
-                                        if (!t) {
-                                                if (has_argument)
-                                                        free(*i);
-                                                continue;
-                                        }
-                                }
-
-                                *(j++) = *i;
+                /* See if we shall mount this together with another controller? */
```

Comment:
nit: This reads weird, it shouldn't have a question mark (or remove "See if we shall")